### DISCUSSION TITLE
怎么用新的store替换旧的store

### DISCUSSION THREAD

**Discussion by battleingame** [vuejs/pinia#2308] [Help and Questions]
框架： vue3 + vite4.2+ qiankun2.0 +pinia
前提：使用微前端框架，需要共享状态
现状：父子应用定义了store（子应用定义主要保持单应用信息取值和父应用一致）我将父应用store传递过来，怎么配置或者用什么方法可以达到父子应用用同一个store,而不是子应用再获取一遍数据处理自己的store

望各位大神提供建议

**Comment by 0131LWG** [comment]
我也在思考这个问题，楼主有方案了吗

**Reply by battleingame** [reply]
并没有，还是要重新创建一份

**Comment by stanleyxu2005** [comment]
如果父子应用都可以修改state，直接重用同一个store。
如果初始状态下，子应用希望拿到父应用的数据。clone吧。

**Reply by battleingame** [reply]
克隆后丢失了响应性，父数据改变，子克隆过来的数据并没有改变，这样还是两套数据
